### PR TITLE
feat(memh5): implement .copy() method

### PR DESCRIPTION
**Pull request by leungcalvin**
Implements a `.copy()` method for memh5 objects that makes a deep copy of datasets. Currently does not support nested groups, or non-numpy arrays as datasets.

### CODE REVIEW

File: caput/memh5.py [hunk 1]
```diff
@@ -1774,6 +1796,41 @@ def save(
...
+        # the recursive function that does the moving of stuff from f to g
+        def _copy(f, g):
+            copyattrs(f.attrs, g.attrs, convert_strings=True)
+            for key in f.keys():
+                if is_group(f[key]):
+                    g_key = g.create_group(key)
+                    _copy(f[key], g_key)
+                else:
+                    try:
+                        g.create_dataset(
```
[the remark this answers is not in the document]
Comment (leungcalvin) [reply caput/memh5.py]: Can you show me where to look to figure out how to write this?

**Comment by leungcalvin** [comment]
Apologies for the last commit, I was careless with a `git pull` and ended up unknowingly having two implementations of `copy` floating around. I have refactored everything to work for `MemDiskGroup`, and I've implemented the `shared` keyword.

**Comment by leungcalvin** [comment]
Do either of you know what is the source of the `readthedocs` error? @jrs65 @kiyo-masui

**Comment by kiyo-masui** [comment]
I do not know where the error is from. However, `readthedocs` parses the doc strings, so its possible they aren't formatted correctly (it expects a very particular formatting conversion).

**Comment by ketiltrout** [comment]
It's apparently a Sphinx bug: https://github.com/sphinx-doc/sphinx/issues/9817 which is fixed in Sphinx 4.3, but we've pinned Sphinx to version 3.4.2.

**Comment by leungcalvin** [comment]
@zkader and I would like to push this soon. Is the sphinx bug not an issue for the main branch of `caput`?

**Comment by ljgray** [comment]
I'm going to close this as this functionality was added in #236, based on feedback from this PR

**Pull request closed without merging**